extensions/amp-accordion/1.0/accordion.js

```diff
@@ -397,16 +415,47 @@ export function AccordionContent({
 }) {
   const ref = useRef(null);
   const hasMountedRef = useRef(false);
-  const {contentId, headerId, expanded, animate, setContentId} = useContext(
-    SectionContext
-  );
+  const {
+    contentId,
+    headerId,
+    expanded,
+    animate,
+    setContentId,
+    expandHandler,
+    experimentDisplayLocking,
+  } = useContext(SectionContext);
   const classes = useStyles();
+  const [contentVisibility, setContentVisibility] = useState(false);
+  const useDisplayLocking = contentVisibility && experimentDisplayLocking;
 
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const element = ref.current;
+    const win = element.ownerDocument.defaultView;
+    const supportsContentVisibility = win.CSS.supports(
+      'content-visibility',
+      'hidden-matchable'
+    );
+    setContentVisibility(supportsContentVisibility);
+    if (!experimentDisplayLocking || !supportsContentVisibility) {
+      return;
+    }
+    const beforeMatchHandler = () => {
+      if (element.hasAttribute('hidden')) {
+        expandHandler();
+      }
+    };
+    element.addEventListener('beforematch', beforeMatchHandler);
+    return () => element.removeEventListener('beforematch', beforeMatchHandler);
+  }, [expandHandler, experimentDisplayLocking]);
+
   useLayoutEffect(() => {
```

Comment:
What does this block do?

1. Checks that the current environment supports the CSS property `'content-visibility' : 'hidden-matchable'`.  If the browser does not support this, we must exit and not support the feature.  Then we set a state variable within the component to track.

2. If the CSS property is supported AND top-level prop is true, then we add an eventListener to listen for `ctrl + f` finds.  The event listener will cause the content section to expand if it hears anything (AND the component is currently collapsed).